server/src/test/java/io/crate/lucene/BooleanEqQueryTest.java

```diff
@@ -107,4 +107,15 @@ public void test_BooleanEqQuery_TermsQuery() {
         assertThat(query.getClass().getName()).endsWith("SortedNumericDocValuesSetQuery");
         assertThat(query).hasToString("arr2: [0, 1]");
     }
+
+    @Test
+    public void test_not_on_boolean_column() {
+        Query query = convert("not(a1)");
+        assertThat(query).isExactlyInstanceOf(BooleanQuery.class);
+        assertThat(query).hasToString("+(+*:* -a1:T) +FieldExistsQuery [field=a1]");
+
+        query = convert("not(a2)");
+        assertThat(query).isExactlyInstanceOf(BooleanQuery.class);
+        assertThat(query).hasToString("+(+*:* -a2:[1 TO 1]) +FieldExistsQuery [field=a2]");
```

Comment:
```suggestion
        Query query = convert("not(a1)");
        assertThat(query).isExactlyInstanceOf(BooleanQuery.class);
        assertThat(query)
            .as("Uses term query and fields exist if index is available")
            .hasToString("+(+*:* -a1:T) +FieldExistsQuery [field=a1]");

        query = convert("not(a2)");
        assertThat(query).isExactlyInstanceOf(BooleanQuery.class);
        assertThat(query)
            .as("Uses DocValuesRangeQuery and FieldsExist if index is not available")
            .hasToString("+(+*:* -a2:[1 TO 1]) +FieldExistsQuery [field=a2]");
```

To make clarify the expected output